### DISCUSSION TITLE
Can't append to a file with file.write. What am I doing wrong?

### DISCUSSION THREAD

**Discussion by Radio-biscuit** [savonet/liquidsoap#4048] [Q&A]
Hello,

I'm trying to set up a playlist log using file.write (Windows 2.2.4) and I can't seem to stop new data overwriting the original data.

If I use:

```liquidsoap
file.write(append=true,data="Test one\n","testing.log")
file.write(append=true,data="Next test\n","testing.log")
```

The file contains just "Next test" when I open it in a text editor. Is this what's supposed to happen or am I missing something? I had a look for answers, but it doesn't seem that anyone else is getting this problem. I feel I'm missing something somewhere.

Thanks, A.

**Comment by toots** [comment]
Hi!

Thank. I can confirm the issue by running the windows binary with `wine`. The problem does not seem to occur on linux.

Still investigating more, will report.

**Comment by Radio-biscuit** [comment]
Ah! I shall try on my Linux kit.

I know that Windows has a very different file system and I have both
systems to test on. My "production" machine is Linux, so I will try there
now.

Thanks again! A.

**Comment by Radio-biscuit** [comment]
Yes, I can confirm that this works with no problems on Linux (Ubuntu 24.04).

I shall be using Linux, so I will work on that machine for now. I use the
Windows machine for testing scripts before using on "production". So I can
also test on the "production" machine using a different instance.

Thanks,

A.

**Reply by toots** [reply]
Great. The problem has been reported and fixed on the compiler: https://github.com/ocaml/ocaml/pull/13326. I'll make sure our next released binary has the patch backported as well.
